packages/subgraph/src/v1-analytics/mapping.ts

```diff
@@ -291,8 +332,52 @@ export function handleTransactionCancelled(event: TransactionCancelled): void {
       assetBalance.assetId = transaction.receivingAssetId;
       assetBalance.router = event.params.router.toHex();
       assetBalance.amount = new BigInt(0);
+      assetBalance.supplied = new BigInt(0);
+      assetBalance.locked = new BigInt(0);
```

Comment:
`assetBalance.locked` will be set to `-transaction.amount` in this case.

the below `assetBalance.locked = assetBalance.locked.sub(transaction.amount);` gets executed after this block...

Reply:
is that expected behavior?

Reply:
unless it's like, an unsigned int and cant below 0 lol

Reply:
it shouldn't ever fall into the case where its not defined, ill use a force unwrap to show that instead.